test/rpc/test_rpc_schema.py

```diff
@@ -0,0 +1,320 @@
+            },
+            "latest",
+        ],
+        [
+            "latest",
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": "0x01",
+                "calldata": [],
+            },
+        ],
+    ),
+)
+def test_schema_raises_on_invalid_args(params):
+    """
+    Test params validation with args
+    """
+    resp = rpc_call("starknet_call", params=params)
+
+    assert "error" in resp
+    error = resp["error"]
+
+    assert error["code"] == RpcErrorCode.INVALID_PARAMS.value
+
+
+@pytest.mark.usefixtures("run_devnet_in_background")
+@pytest.mark.parametrize(
+    "run_devnet_in_background",
+    [["--disable-rpc-request-validation"]],
+    indirect=True,
+)
+def test_schema_does_not_raise_on_disabled_request_validation():
+    """
+    Test schema validation is disabled by CLI argument
+    """
+    resp = rpc_call(
+        "starknet_call",
+        params={
+            "block_id": "latest",
+            "request": {
+                "contract_address": 1234,
+                "entry_point_selector": -1,
+                "calldata": ["a", "b", "c"],
+            },
+        },
+    )
+
+    # Error will be raised when trying to execute function, but it shouldn't be the INVALID_PARAMS error
+    error = resp["error"]
+    assert error["code"] != RpcErrorCode.INVALID_PARAMS.value
+
+
+@pytest.mark.usefixtures("run_devnet_in_background")
+@pytest.mark.parametrize(
+    "params",
+    (
+        {
+            "request": {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            "block_id": "latest",
+        },
+        {
+            "block_id": "latest",
+            "request": {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+        },
+        {
+            "request": {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            "block_id": {"block_hash": "0x00"},
+        },
+        {
+            "request": {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            "block_id": {
+                "block_number": 0,
+            },
+        },
+    ),
+)
+def test_schema_does_not_raise_on_correct_kwargs(params):
+    """
+    Test kwargs validation allows valid requests
+    """
+
+    resp = rpc_call("starknet_call", params=params)
+
+    # Error will be raised because address is correctly formatted but incorrect
+    error = resp["error"]
+    assert all(error["code"] != code.value for code in RpcErrorCode)
```

Comment:
Shouldn't it return some expected error? Something that we can assert? If I understand this correctly, we are just asserting the received error code is not one of the predefined RPC error codes?

Reply:
Yes, it is meant to validate if the error returned was not one of the "internal" RPC errors i.e. not defined in the spec itself.

Reply:
Ok, but what kind of type of error will be raised? Will it result in an internal server error? If not, why can't we assert that actual error?

Reply:
It is supposed to raise one of the errors documented in rpc for `starknet_call` (assuming we are doing error handling correctly). I can assert it's one of these errors like I did in the method above if you think it's useful.

What is most important in this tests is that no `RpcErrorCode`s are raised - input validated correctly and `call` method actually run.